### PR TITLE
Forwarder buy shipment and carrier share location #6

### LINKED ISSUE #6: Forwarder buy offer and carrier share location

### PR DESCRIPTION
**Pull request by matzayonc**
resolves #6

### CODE REVIEW

File: programs/protocol/src/data/carrier.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+pub struct Availability {
+    pub time: u64, // when 0, carrier is not available
+    pub location: GeoLocation,
+}
+
+#[account(zero_copy)]
+#[derive(Debug, Default, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct Carrier {
+    pub authority: Pubkey,
+    pub availability: Availability,
```
Comment (tommysr) [inline programs/protocol/src/data/carrier.rs:19]: just one?

**Review by tommysr — COMMENTED** [review]
Looks good but only if this buy option is supposed to be placed after bidding and accepting by shipper. But imo buying is not proper term to use.

File: programs/protocol/src/data/carrier.rs (hunk 1 above)
Comment (matzayonc) [reply programs/protocol/src/data/carrier.rs:19]: All the future availability would depend on the next shipment

We can add longer-term planning later, but for now I think the map with available time is more than enough

**Comment by matzayonc** [comment]
> Looks good but only if this buy option is supposed to be placed after bidding and accepting by shipper. But imo buying is not proper term to use.


I think this would be an instant snatch from the market, accepting the original price, possibly given some constraints in the future (like a whitelist).

Submitting an offer, and accepting it would be a rare case when the forwarder wants to haggle.

**Review by tommysr — APPROVED** [review]

**Pull request merged**